src/api/java/mcp/mobius/waila/api/component/NamedItemComponent.java

```diff
@@ -0,0 +1,62 @@
+package mcp.mobius.waila.api.component;
+
+import mcp.mobius.waila.api.ITooltipComponent;
+import mcp.mobius.waila.api.WailaHelper;
+import mcp.mobius.waila.api.__internal__.ApiSide;
+import mcp.mobius.waila.api.__internal__.IApiService;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.ItemLike;
+
+/**
+ * Component that renders an {@link ItemStack} with its name.
+ */
+@ApiSide.ClientOnly
+public class NamedItemComponent implements ITooltipComponent {
+
+    public static final NamedItemComponent EMPTY = new NamedItemComponent(ItemStack.EMPTY);
+
+    private static final Font FONT = Minecraft.getInstance().font;
+    public static final int HEIGHT = FONT.lineHeight;
```

Comment:
Don't get the client instance in a static context, this will crash on the server when the class is initialized. Just get it every time it is needed.

Reply:
Didn't know that, I replaced it with `getFont` method similar to one in `WrappedComponent`